packages/eslint-plugin-shopify/lib/config/rules/stylistic-issues.js

```diff
@@ -21,6 +21,8 @@ module.exports = {
   'eol-last': 'warn',
   // Disallow space between function identifier and application
   'func-call-spacing': 'warn',
+  // Require function names to match the name of the variable or property to which they are assigned
+  'func-name-matching': 'warn',
```

Comment:
@lemonmade How do you feel about this rule? http://eslint.org/docs/rules/func-name-matching